internal/job/vault-cred-sync.go

```diff
@@ -38,7 +39,14 @@ type ServiceCredentail struct {
 	Password       string            `json:"password"`
 	AdditionalData map[string]string `json:"additionalData"`
 }
-
+type GenericCredential struct {
+	CredentialType  string            `json:"credentialType"`
+	EntityName      string            `json:"entityName"`
+	CertIndentifier string            `json:"certIndetifier"`
+	ClusterId       string            `json:"clusterId"`
+	Token           string            `json:"token"`
+	Data            map[string]string `json:"additionalData"`
```

Comment:
change Data to "Credential"

remove ClusterId, Token, these will go under Credential" map[string]string